### ISSUE TITLE
Context cancellation does not close file selection dialogs immediately on Windows

### ISSUE THREAD

**Issue by gz-95** [ncruces/zenity#119]
Thanks for this really helpful library! Just having some trouble with closing the dialogs programmatically.

It seems that when the context is canceled, the dialog remains open until the user interacts with it e.g. navigating different directories etc. Here's an example code that reproduces the issue on Windows 10.0.19045 Build 19045

```
package main

import (
	"context"
	"fmt"

	"github.com/ncruces/zenity"
)

func main() {
	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
	defer cancel()

	stop := context.AfterFunc(ctx, func() {
		fmt.Println("canceled!")
	})

	defer stop()

	file, err := zenity.SelectFile(
		zenity.Context(ctx),
	)

	if err != nil {
		panic(err)
	}

	fmt.Println("select", file)
}
```

Basically, the call to `zenity.SelectFile` is blocked with no way of returning even if the context has expired unless the user is actively interacting with the underlying dialog.

**Comment by ncruces** [comment]
I don't have a windows machine handy.
Do you know which specific windows API is blocked?

**Comment by gz-95** [comment]
In my testing, it appears that [`IModalWindow::Show`](https://github.com/ncruces/zenity/blob/3a348ffda876ba50ad17d945ff69ffa7155614cb/file_windows.go#L261) continues to block until active user interaction even though a successful call to [`IFileDialog::Close`](https://github.com/ncruces/zenity/blob/3a348ffda876ba50ad17d945ff69ffa7155614cb/file_windows.go#L255) has returned.

I'm starting to wonder if this is just the way these Windows functions are implemented?

Also, it looks like [`dialogHookProc`](https://github.com/ncruces/zenity/blob/3a348ffda876ba50ad17d945ff69ffa7155614cb/util_windows.go#L130) never gets called even with a successful `SetWindowsHookEx` registration. That explained why I wasn't seeing this [line](https://github.com/ncruces/zenity/blob/3a348ffda876ba50ad17d945ff69ffa7155614cb/util_windows.go#L165) firing.

**Comment by ncruces** [comment]
Thanks for the detailed report. Yeah, I'm not seeing a way out of this…

**Comment by gz-95** [comment]
Agreed. I've ditched `IFileDialog::Close` because I'm convinced that it was only ever intended to be called inside `IFileDialogEvents` callbacks.

Workaround: if I call `CoInitializeEx` with `COINIT_APARTMENTTHREADED`, somehow `SetWindowsHookEx` seems to work again for the dialogs. So I went with that and just call `PostMessage(hwnd, WM_SYSCOMMAND, SC_CLOSE, 0)` when the context expires.

**Comment by ncruces** [comment]
This was put in place because of #89. I'm a bit reluctant to go back on that.
